src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -479,9 +479,27 @@ def show_service_principal(client, identifier):
     return client.get(object_id)
 
 
-def delete_service_principal(client, identifier):
-    object_id = _resolve_service_principal(client, identifier)
-    client.delete(object_id)
+def delete_service_principal(identifier):
```

Comment:
Deleting a service principal and leaving the application behind in the tenant is not a scenario in rbac, not very meaningful as well in general AAD auth world. The application is useless once the SP is gone. That is why in portal, AAD UI will automatically create a SP when you author an application. 
For multi-tenants scenario(not related with rbac) that the SP was created for an external application (through admin consent), the code change here will only delete the SP. The application still stays in the original tenant.